Added more Integration Tests and automation for running Integration Tests locally (#117)
gardener/autoscaler: cluster-autoscaler/integration/usage.md

	cd ./cluster-autoscaler
	```

2. Export the following environment variables
	```
	export CONTROL_NAMESPACE=<Shoot namespace in the Seed>
	export TARGET_KUBECONFIG=<Path to the kubeconfig file of the Shoot>
	export CONTROL_KUBECONFIG=<Path to the kubeconfig file of the Seed (or the control plane where the Cluster Autoscaler & Machine deployment objects exists)>
	export KUBECONFIG=<Path to the kubeconfig file of the Shoot>
	export volume_zone=<zone in cluster with zero nodes>

> 	export cluster_provider=<aws/gcp/azure/...>

Reviewer: ```suggestion
	export VOLUME_ZONE=<zone with zero nodes where the PV needs to be created to perform test>
	export PROVIDER=<aws/gcp/azure/...>
```

	```
	

> 2. Alternatively, you could use the `make download-kubeconfig` to download and place the kubeconfigs in dev folder and follow the steps as mentioned in output of the 			command. (only to be used when working with gardenctl V2 and both the clusters are on gardener)

Reviewer: ```suggestion
2. Alternatively, you could use the `make download-kubeconfigs` to download and place the kubeconfigs in dev folder and follow the steps as mentioned in output of the 			command. (only to be used when working with gardenctl V2 and both the clusters are on gardener)
```

		```

	</details>

## Tests Covered

1. Deploy new workload that asks for more resources and it should create a new machine to accommodate that.
2. Scaling to 3 instances of the above workload to increase the number of machines to 3.

> 3. Removing all the above instances to remove all the newly added machines.

Reviewer: ```suggestion
3. Removing all the above workload to remove all the newly added machines.
```

> 4. Should not scale up above the max limit mentioned in the dashboard for machines.

Reviewer: ```suggestion
4. Should not scale up above the max limit for the nodegrp
```

> 5. Should not scale lower than the min limit mentioned in the dashboard for machines.

Reviewer: ```suggestion
5. Should not scale lower than the min limit for the nodegrp
```

6. Should scale up on the basis of taints and not only on workload size.
7. Should respond by shifting load if some taint is removed from a workload/node.

> 8. AutoScaler scales up in the zone in which storageClass is available is already present

Reviewer: ```suggestion
8. AutoScaler scales up a node in zone where PV already present and pod is requesting that PV. CSI PV is used.
```

> 9. Should not scale down a node with  annotation cluster-autoscaler.kubernetes.io/scale-down-disabled": "true"

Reviewer: ```suggestion
9. Should not scale down a node with  annotation "cluster-autoscaler.kubernetes.io/scale-down-disabled": "true"
```
